enzyme/Enzyme/EnzymeLogic.cpp

```diff
@@ -3410,19 +3410,21 @@ Function *EnzymeLogic::CreatePrimalAndGradient(
           replacedReturns[orig] = si;
         }
 
-        if (dretAlloca && !gutils->isConstantValue(orig->getReturnValue())) {
-          rb.CreateStore(gutils->invertPointerM(orig->getReturnValue(), rb),
+        if (key.retType == DIFFE_TYPE::DUP_ARG || 
+	    key.retType == DIFFE_TYPE::DUP_NONEED) {
+        	if (dretAlloca) {
+          		rb.CreateStore(gutils->invertPointerM(orig->getReturnValue(), rb),
                          dretAlloca);
-        }
-        if (key.retType == DIFFE_TYPE::OUT_DIFF) {
+		}
+        } else if (key.retType == DIFFE_TYPE::OUT_DIFF) {
           assert(orig->getReturnValue());
           assert(differetval);
           if (!gutils->isConstantValue(orig->getReturnValue())) {
```

Comment:
If it is constant, this should be set to zero.

Reply:
Not quite, the if the return is active this should still be set to differetval, but if it is a constant value then it should be set to zero